packages/json-schema/test/integration/build-schema.test.ts

```diff
@@ -0,0 +1,159 @@
+        nul: {type: 'null'},
+      });
+      expect(fooJSON.properties).to.not.containEql({
+        undef: {type: 'undefined'},
+      });
+    });
+    it('does not convert properties that have not been decorated', () => {
+      @model()
+      class NoPropertyMeta {
+        prop: string;
+      }
+      @model()
+      class OnePropertyDecorated {
+        @property() foo: string;
+        bar: boolean;
+        baz: number;
+      }
+      expect(modelToJSONDef(NoPropertyMeta)).to.eql({});
+      expect(modelToJSONDef(OnePropertyDecorated)).to.deepEqual({
+        properties: {
+          foo: {
+            type: 'string',
+          },
+        },
+      });
+    });
+    it('does not convert models that have not been decorated with @model()', () => {
+      class Empty {}
+      class NoModelMeta {
+        @property() foo: string;
+        bar: number;
+      }
+      expect(modelToJSONDef(Empty)).to.eql({});
+      expect(modelToJSONDef(NoModelMeta)).to.eql({});
+    });
+    it('properly converts to JSON definition', () => {
+      expect(fooJSON).to.deepEqual({
+        properties: {
+          str: {
+            type: 'string',
+          },
+          num: {
+            type: 'number',
+          },
+          bool: {
+            type: 'boolean',
+          },
+          obj: {
+            type: 'object',
+          },
+          arrStr: {
+            type: 'array',
+            items: {
+              type: 'string',
+            },
+          },
+          bar: {
+            $ref: '#definitions/Bar',
+          },
+          arrBar: {
+            type: 'array',
+            items: {
+              $ref: '#definitions/Bar',
+            },
+          },
+        },
+      });
+    });
+    describe('toJSONProperty', () => {
+      // tslint:disable-next-line:no-any
+      let meta: any = ModelMetadataHelper.getModelMetadata(Foo);
+      let propMeta = meta.properties;
+      it('converts primitively typed property correctly', () => {
+        expect(toJSONProperty(propMeta['str'])).to.deepEqual({
+          type: 'string',
+        });
+        expect(toJSONProperty(propMeta['num'])).to.deepEqual({
+          type: 'number',
+        });
+        expect(toJSONProperty(propMeta['bool'])).to.deepEqual({
+          type: 'boolean',
+        });
+      });
+      it('converts object property correctly', () => {
+        expect(toJSONProperty(propMeta['obj'])).to.deepEqual({
+          type: 'object',
+        });
+      });
+      it('converts customly typed property correctly', () => {
+        expect(toJSONProperty(propMeta['bar'])).to.deepEqual({
+          $ref: '#definitions/Bar',
+        });
+      });
+      it('converts arrays correctly', () => {
+        expect(toJSONProperty(propMeta['arrStr'])).to.deepEqual({
+          type: 'array',
+          items: {
+            type: 'string',
+          },
+        });
+        expect(toJSONProperty(propMeta['arrBar'])).to.deepEqual({
```

Comment:
One (logical) assert per test please. There should be two tests here - one for string arrays, another for Bar arrays.